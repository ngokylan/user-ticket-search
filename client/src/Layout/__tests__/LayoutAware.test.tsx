import React from 'react';
import { mount, shallow } from 'enzyme';
import withLayoutAware from '../component/LayoutAware';

describe('Layout::LayoutAware', () => {
  describe('api', () => {
    it('should mount properly', () => {
      jest.clearAllMocks();
      const SampleComponent = () => <div>content</div>;
      const LayoutAwareSampleComponent = withLayoutAware(SampleComponent);
      const wrapper = mount(<LayoutAwareSampleComponent />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should shallow properly', () => {
      jest.clearAllMocks();
      const SampleComponent = () => <div>content</div>;
      const LayoutAwareSampleComponent = withLayoutAware(SampleComponent);
      const wrapper = shallow(<LayoutAwareSampleComponent />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
