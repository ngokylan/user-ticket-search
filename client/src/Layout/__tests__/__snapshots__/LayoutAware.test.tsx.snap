// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout::LayoutAware api should mount properly 1`] = `
<div>
  content
</div>
`;

exports[`Layout::LayoutAware api should shallow properly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
